server/aesdsocket.c

```diff
@@ -22,49 +22,331 @@
+               
+         
+                len3= read(fd, read_buf, offset);
+                if(len3 == -1)
+                {
+                        perror("Read unsuccessful\n");
+                       func_close();   
+                       free(read_buf);
+         		free(rec_buf);
         
-        close(ret);
-        if(ret4 != -1)
+                }
+                
+                
+                
+                sent+=len3;   
+                
+                //stat("/var/tmp/aesdsocketdata.txt", &st);
+                
+                
         
-        if(close(ret4) == -1)
+        ret5= send(threadlocal->accept_fd, read_buf, len3, 0);
+         if(ret5 == -1)
+         {
+                perror("Send unsuccessful\n");
+                func_close(); 
+                free(read_buf);
+          	free(rec_buf);
+         
+          }
+          
+          
+        
+       //syslog(LOG_USER, " read rec_buf = %s", read_buf);
+        
+        pthread_mutex_unlock(&mutex);
+        
+         if (sigprocmask(SIG_UNBLOCK,&threadlocal->mask,NULL) == -1) {   
+               perror("unblock failed");
+               }
+        
+       ret5 = close(threadlocal->accept_fd);
+        if(ret5 == -1)
          {
                 perror("close socket unsuccessful\n");
-                                
-          } else syslog(LOG_USER, "Closed connection from %s",ip);
+                free(read_buf);
+          	free(rec_buf);
+
+                
+                                 
+          }else
+          syslog(LOG_USER, "Closed connection from %s",ip);
+          
+          threadlocal->thread_complete_success=1;
+          
+          free(read_buf);
+          free(rec_buf);
+
+
+}
+
+void thread_for_timer( union sigval sigval)
+{
+	
+	
+	
+	char timer_buf[timer_bufsize];
+	
+	time_t time_cur;
+	time(&time_cur);
+	
+	
+	struct tm *time_stamp;
+	
+	time_stamp= localtime(&time_cur);
+	
+	int time_buf_siz= strftime(timer_buf, timer_bufsize, "timestamp:%Y %b %a %d %H:%M:%S%n",time_stamp) ;
+	if(time_buf_siz < 0 ) perror(" strftime failed");
+	
+	pthread_mutex_lock(&mutex);
+	timestamp_len = time_buf_siz;
+	
+	if( write(fd, timer_buf, time_buf_siz) == -1)
+	{
+		perror("timetsamp write error");
+		func_close();
+	}
+	
+	pthread_mutex_unlock(&mutex);
+
+}
+
+
+void sig_handler(int signum)
+{
+        
+        signal_bool=1;
+        syslog(LOG_USER,"Caught signal exiting...");
         
-         free(rec_buf);
-         free(read_buf);
+        
+        //close(ret);
+        
+        shutdown(ret, SHUT_RDWR);
          
-         remove("/var/tmp/aesdsocketdata.txt");
+         //remove("/var/tmp/aesdsocketdata.txt");
          
-        if(close(fd) == -1) 
-        {
-                syslog(LOG_ERR,"close file unsuccessful\n");
-                
-        }
+        // func_close();       
+     
         
-        closelog();
+        //closelog();
         
-        exit(1);
+       // exit(1);
```

Comment:
@chaisuresh delete old code rather than comment out, use revision control to revert later if needed.